src/compiler/transformers/declarations/diagnostics.ts

```diff
@@ -767,7 +767,7 @@ export function createGetIsolatedDeclarationErrors(resolver: EmitResolver): (nod
         if (isSetAccessor(node.parent)) {
             return createAccessorTypeError(node.parent);
         }
-        const addUndefined = resolver.requiresAddingImplicitUndefined(node, /*enclosingDeclaration*/ undefined);
+        const addUndefined = resolver.requiresAddingImplicitUndefined(node, node.parent);

```

Comment:
I was initially worried about this just because `enclosingDeclaration` is weird and should/shouldn't be certain kinds of nodes, but in this case, `node.parent` is a function declaration (or expression), so is a valid choice.

Reply:
(also, no other caller than this ever passes `undefined` into `requiresAddingImplicitUndefined` explicitly, so, that's of note)